### PR TITLE
update Logging Interceptor to log connection failures

### PR DESCRIPTION
**Pull request by adennie**
BEFORE

05-06 08:21:41.939 10449-11016/com.blah.blah D/OkHttp: --> GET https://somewhere.com/ http/1.1
05-06 08:22:11.205 10449-11016/com.blah.blah D/libc: [NET] android_getaddrinfofornetcontext+,hn 24(0x736f6c6f2d7175),sn(),hints(known),family 0,flags 1024, proc=com.blah.blah
05-06 08:22:11.207 10449-11016/com.blah.blah D/libc: [NET] android_getaddrinfo_proxy get netid:0
05-06 08:22:11.557 10449-11016/com.blah.blah D/libc: [NET] android_getaddrinfo_proxy-, NODATA

AFTER

05-10 09:10:42.693 17637-19160/com.blah.blah D/OkHttp: --> GET https://somewhere.com/ http/1.1
05-10 09:10:42.697 17637-19160/com.blah.blah D/libc: [NET] android_getaddrinfofornetcontext+,hn 25(0x736f6c6f2d7175),sn(),hints(known),family 0,flags 1024, proc=com.blah.blah
05-10 09:10:42.697 17637-19160/com.blah.blah D/libc: [NET] android_getaddrinfo_proxy get netid:0
05-10 09:10:43.202 17637-19160/com.blah.blah D/libc: [NET] android_getaddrinfo_proxy-, NODATA
05-10 09:10:43.203 17637-19160/com.blah.blah D/OkHttp: <-- Failed to send: java.net.UnknownHostException: Unable to resolve host "somewhere.com": No address associated with hostname

### CODE REVIEW

**Comment by JakeWharton** [comment]
Can you add a test?

**Comment by adennie** [comment]
Could someone restart the Travis build?  Seems to have had a timeout installing the JDK from Oracle.  Hopefully a transient problem.

File: okhttp-logging-interceptor/src/test/java/okhttp3/logging/HttpLoggingInterceptorTest.java [hunk 1]
```diff
@@ -641,6 +642,26 @@ private void bodyGetNoBody(int code) throws IOException {
         .assertNoMoreLogs();
   }
 
+    @Test public void connectFail() throws IOException {
+        setLevel(Level.BASIC);
+        url = new HttpUrl.Builder()
+            .scheme("http")
+            .host("okhttp-notfound.com")
```
Comment (swankjesse) [inline okhttp-logging-interceptor/src/test/java/okhttp3/logging/HttpLoggingInterceptorTest.java]: sadly when you run tests like this in certain environments, you get a result. I’ve seen it on Comcast. There are other special values that’ll work though . . . lemme look it up.

Comment (JakeWharton) [inline okhttp-logging-interceptor/src/test/java/okhttp3/logging/HttpLoggingInterceptorTest.java]: How about a custom `Dns` instead. Tests should absolutely not be connecting to real things (even real DNS servers).

**Comment by swankjesse** [comment]
Thanks! I fixed a couple tiny nitpicks (indentation! imports!)  and tweaked your message. But otherwise this is great. I wonder if it’s worthwhile to do something similar to this if streaming the response body fails.

**Pull request closed without merging**